test/testsuite/lantest_dircache_stats.c

```diff
@@ -0,0 +1,396 @@
+/*
+ * Copyright (c) 2025, Andy Lemin (andylemin)
+ * Credits; Based on work by Netatalk contributors
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+/* Configuration header (must be first) */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif /* HAVE_CONFIG_H */
+
+/* Standard C library includes */
+#include <stdio.h>
+
+#ifdef __linux__
+
+/* Standard C library includes for Linux implementation */
+#include <errno.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+/* Netatalk includes for INIPARSER_GETSTRDUP macro */
+#include <atalk/globals.h>
```

Comment:
It's pleasing to see that you were able to make good use of the iniparser library!

Putting the iniparser wrapper macros in atalk/globals.h was kind of a last resort. We previously had our own vendored iniparser library so they used to live in the modified iniparser.h. Do you have a suggestion of a better way to organize these?

(Not directly relevant to this PR, but I was reminded about this when I see that you had to import globals.h just for this macro.)

Reply:
Ahh I was confused why importing the iniparser libraries alone was not enough and this was required (before importing iniparser). Took me a while to figure out actually, including adding the iniparser dependancy in meson.build.
Generally config and logging setup/handling code often have startup race conditions, and you can't avoid it that you need to import them first. But there should only be a few things needing this. I usually add logging managers and config handlers into a special startup dependancies library (named startup or something), and nothing else is allowed in there other than things which solve startup race conditions. At least having naming like that is easier to manage.